src/booleans.js

```diff
@@ -1,61 +1,65 @@
-function negate(a) {
-  // your code here
+const negate = (a) => {
+  return !(a);
 };
 
-function both(a, b) {
-  // your code here
+const both = (a, b) => {
+  return (a && b);
 };
 
-function either(a, b) {
-  // your code here
+const either = (a, b) => {
+  return (a || b);
 };
 
-function none(a, b) {
-  // your code here
+const none = (a, b) => {
+  return !(a || b)
 };
 
-function one(a, b) {
-  // your code here
+const one = (a, b) => {
+  return ((a && !b) || (!a && b));
 };
 
-function truthiness(a) {
-  // your code here
+const truthiness = (a) => {
+  if (a) {
+    return true
+  } else {
+    return false
+  }
 };
 
-function isEqual(a, b) {
-  // your code here
+const isEqual = (a, b) => {
+  return (a === b);
 };
 
-function isGreaterThan(a, b) {
-  // your code here
+const isGreaterThan = (a, b) => {
+  return (a > b);
 };
 
-function isLessThanOrEqualTo(a, b) {
-  // your code here
+const  isLessThanOrEqualTo = (a, b) => {
+  return (a <= b);
 };
 
-function isOdd(a) {
-  // your code here
+const isOdd = (a) => {
+  return (a % 2 != 0);
 };
 
-function isEven(a) {
-  // your code here
-};
+const isEven = (a) => {
+  return (a % 2 == 0);
+}
 
-function isSquare(a) {
-  // your code here
+const isSquare = (a) => {
+  return Math.sqrt(a) % 1 === 0;
 };
 
-function startsWith(char, string) {
-  // your code here
+const startsWith = (char, string) => {
+  return string.startsWith(char);
 };
 
-function containsVowels(string) {
-  // your code here
+const containsVowels = (string) => {
+  return string.match('a|e|i|o|u|A|E|I|O|U/i') ? true : false;
```

Comment:
A couple of things with this one:
- the case-insensitive `/i` flag at the end means you dont have to put in both upper and lower case letters to get a match,
- The ternary isnt needed, as the function should return a truthy value so this is just adding a bit of extra noise,
- Theres a very similar solution to this as what you used in the arrays file:
```suggestion
  return /[aeiou]/i.test(string);
```

I wouldnt worry too much about memorising the details about how regex works this early on though. Its one of those things that will get easier the more you do it and become more familiar with the use cases and weird syntax